### DISCUSSION TITLE
Mapping of nested entities does not work

### DISCUSSION THREAD

**Discussion by KristofDeTroch** [mikro-orm/mikro-orm#6239] [General]
### Describe the bug

When fetching nested relations through a complex query (knex) and using the orm to map the data back to the entity to make use of the typings, it does not map any nested entities.

### Reproduction

import {
  Entity,
  MikroORM,
  PrimaryKey,
  Property,
  OneToMany,
  Collection,
  ManyToOne,
  Rel,
  OneToOne,
  Opt,
} from "@mikro-orm/postgresql";

@Entity()
class User {
  @PrimaryKey()
  id!: number;

  @Property()
  name: string;

  @Property({ unique: true })
  email: string;

  @ManyToOne(() => Location)
  public location: Rel<Location>;

  constructor(name: string, email: string) {
    this.name = name;
    this.email = email;
  }
}
@Entity()
class Location {
  @PrimaryKey()
  id!: number;

  @Property({ nullable: true })
  public facility?: string;

  @Property({ unique: true })
  email: string;

  @OneToMany(() => User, (ut) => ut.location)
  public users = new Collection<User>(this);
  @OneToOne(() => Mandate, (m) => m.location, { nullable: true })
  public mandate?: Opt<Mandate>;
}
@Entity()
export class Mandate {
  @PrimaryKey()
  id!: number;
  @Property()
  public teamId: string;

  @OneToOne(() => Location, (l) => l.mandate, { owner: true })
  public location: Rel<Location>;
}

let orm: MikroORM;

beforeAll(async () => {
  orm = await MikroORM.init({
    dbName: ":memory:",
    entities: [User, Location, Mandate],
    debug: ["query", "query-params"],
    allowGlobalContext: true, // only for testing
  });
  await orm.schema.refreshDatabase();
});

afterAll(async () => {
  await orm.close(true);
});

test("basic CRUD example", async () => {
  const user = orm.em.create(User, { name: "Foo", email: "foo" });
  const location = orm.em.create(Location, { facility: "Foo", email: "foo" });
  const mandate = orm.em.create(Mandate, { teamId: "1", location });
  user.location = location;
  await orm.em.flush();
  orm.em.clear();

  const qb = orm.em.createQueryBuilder(User, "ut");

  const knexQb = qb
    .select("*")
    .leftJoinAndSelect("ut.location", "location")
    .leftJoinAndSelect("location.mandate", "mandate")
    .getKnexQuery();

  const rawResult = await knexQb;
  const result = {
    items: rawResult.map((rawUser) => {
      const user = orm.em.map(User, rawUser);
      // This should not be required, but this works
      // const location = this.em.map(
      //   Location,
      //   Object.keys(rawUser).reduce((acc, key) => {
      //     if (key.startsWith('location__')) {
      //       acc[key.replace('location__', '')] = rawUser[key];
      //     }
      //     return acc;
      //   }, {}),
      // );
      return {
        userId: user.id,
        isLocationActive: Boolean(user.location.mandate), //Boolean(rawUser.mandate__id),
        contactedAt: user.contactedAt,
        acceptedAt: user.acceptedAt,
        verifiedAt: user.verifiedAt,
        country: user.country,
        postalCode: user.postal,
        locationId: user.location.id, //rawUser.location__id,
        locationName: user.location.facility, //rawUser.location__facility,
      };
    }),
  };
  expect(rawResult[0].mandate__id).toBe(1);
  expect(result.items.find((item) => item.userId === 1)?.isLocationActive).toBe(
    true
  );
});

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.0

### Node.js version

20.12.2

### Operating system

mac os

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Your problem is that you execute the knex instance instead of using `qb.execute()`, which would handle mapping of the aliased columns to property names and map the flat results to object graph. `em.map` wont understand what to do since it doesnt know anything about what aliases are used and what relations they represent.

This is the part that handles it in `qb.execute`, I guess you could try to call it yourself to do the mapping: 

https://github.com/mikro-orm/mikro-orm/blob/master/packages/knex/src/query/QueryBuilder.ts#L987

Why do you want to go through knex like this?
